net/libfab/libfab.c

```diff
@@ -727,19 +726,19 @@ static void libfab_straddr_gen(struct m0_fab__conn_data *cd, char *buf,
 {
 	libfab_ep_ntop(cd->fcd_netaddr, en);
 	if (cd->fcd_tmid == 0xFFFF)
-		sprintf(buf, "%s@%s:12345:%d:*",
+		snprintf(buf, len, "%s@%s:12345:%d:*",
 			cd->fcd_iface == FAB_LO ? "0" : en->fen_addr,
 			cd->fcd_iface == FAB_LO ? "lo" : 
 				((cd->fcd_iface == FAB_TCP) ? "tcp" : "o2ib"),
 			cd->fcd_portal);
 	else
-		sprintf(buf, "%s@%s:12345:%d:%d",
+		snprintf(buf, len, "%s@%s:12345:%d:%d",
 			cd->fcd_iface == FAB_LO ? "0" : en->fen_addr,
 			cd->fcd_iface == FAB_LO ? "lo" : 
 			((cd->fcd_iface == FAB_TCP) ? "tcp" : "o2ib"),
 			cd->fcd_portal, cd->fcd_tmid);
 
-	M0_ASSERT(len >= strlen(buf));
+	M0_ASSERT(len > strlen(buf));
```

Comment:
I'm not sure if my suggestion fits into the scope of this patch, but what do you think about the following idea?

```c
int rc;
rc = snprintf(buf, len, ...);
...
M0_ASSERT(rc > 0);
M0_ASSERT_INFO(rc < len, "Net addr should not be truncated"
                            "(wanted %d, available %d)", rc, (int) len);
```

Reply:
I like your idea. And good catch of the bug!

But you probably meant `M0_ASSERT_INFO(rc >= len, ...)`?

Added. Thanks!

Reply:
I meant `rc < len`. As per the man page: "Thus, a return value of `size` or more means that the output was truncated". If we want to avoid truncation then we must ensure that the return value is less that `size`. Therefore, we must ensure that `rc < len`.

Reply:
Ah, you are right. Fixed, thanks!